scanomatic/ui_server/status_api.py

```diff
@@ -1,63 +1,72 @@
-from flask import Flask, jsonify
+from flask import Blueprint, g, jsonify
 
-from ..io.rpc_client import _ClientProxy
+from ..io.rpc_client import get_client
 from .general import convert_path_to_url, json_abort
 
 
-def add_routes(app: Flask, rpc_client: _ClientProxy):
+blueprint = Blueprint('status_api', __name__)
 
-    @app.route("/api/status/<status_type>")
-    @app.route("/api/status/<status_type>/<status_query>")
-    def _status_api(status_type="", status_query=None):
 
-        if status_type == 'server':
-            if rpc_client.online:
-                return jsonify(success=True)
-            else:
-                return json_abort(
-                    503,
-                    reason="System not yet ready, probably starting up",
-                )
-        elif not rpc_client.online:
-            return jsonify(success=False, reason="Server offline")
-        elif status_type == 'queue':
-            return jsonify(queue=rpc_client.get_queue_status())
-        elif status_type == 'scanners':
-            if status_query is None or status_query.lower() == 'all':
-                return jsonify(scanners=rpc_client.get_scanner_status())
-            elif status_query.lower() == 'free':
-                return jsonify(
-                    scanners={
-                        s['socket']: s['scanner_name']
-                        for s in rpc_client.get_scanner_status()
-                        if 'owner' not in s or not s['owner']
-                    },
+@blueprint.before_request
+def get_rpc():
+    if 'rpc' not in g:
+        g.rpc = get_client()
+
+
+@blueprint.route("/server")
+def status_server():
+    if g.rpc.online:
+        return jsonify(**g.rpc.get_status())
```

Comment:
is `g.rpc` correctly typed, if not, then I don't think it's sufficently beneficial to do get_client() in the before_request.

Reply:
Don't understand what you what me to do? call get_client() in each route?

Reply:
yes rather that if `rpc` isn't typed on `g.rpc`. `get_client` is meant to be caching so should not be costly and if it was a cost issue it's just a matter of slapping a `lru_cache` on it

Reply:
or alternatively should `g.rpc` not be propertly typehinted it can be cast each use.